include/wil/Tracelogging.h

```diff
@@ -1299,15 +1299,17 @@ class ActivityBase : public details::IFailureCallback
 #define __WI_TraceLoggingWriteStart(activity, name, ...) \
     __pragma(warning(push)) __pragma(warning(disable : 4127)) do \
     { \
-        _tlgActivityDecl(activity) static const UINT64 _tlgActivity_Keyword = _tlgActivityRef(activity).Keyword; \
-        static const UINT8 _tlgActivity_Level = _tlgActivityRef(activity).Level; \
-        static const UINT64 _tlgActivityPrivacyTag = _tlgActivityRef(activity).PrivacyTag; \
+        _tlgActivityDecl(activity); \
+        using _tlg_Activity_t = wistd::remove_reference_t<decltype(activity)>; \
+        static constexpr const UINT64 _tlgActivity_Keyword = _tlg_Activity_t::Keyword; \
+        static constexpr const UINT8 _tlgActivity_Level = _tlg_Activity_t::Level; \
+        static constexpr const UINT64 _tlgActivityPrivacyTag = _tlg_Activity_t::PrivacyTag; \

```

Comment:
This is probably the riskiest part of this change, however I'd like to believe that any place where `(*this).Name` works, `decltype(*this)::Name` should also work. Any non-constant should trigger an error in the `static_assert`, even for MSVC